llama.cpp

```diff
@@ -12446,6 +12447,10 @@ static ggml_type llama_tensor_get_type(quantize_state_internal & qs, ggml_type n
             if (arch == LLM_ARCH_FALCON || nx % QK_K != 0) {
                 new_type = GGML_TYPE_Q8_0;
             }
+            else if (ftype == LLAMA_FTYPE_MOSTLY_IQ1_XS) {
+                if (qs.model.hparams.n_expert >= 2) new_type = GGML_TYPE_Q5_K;
```

Comment:
Interesting that you always prefer `IQ4_XS` to `Q4_K`. For a large model (e.g., 70B), the 0.25 bpw saving in `output.weight` is completely negligible, but at least for the models that I tend to use for testing `Q4_K` produces a slightly better result than `IQ4_XS`.

Reply:
Duly noted, I observed a significant quality drop from output Q5_K to IQ4_XS on Mixtral, and I wondered if the IQ quant was the problem beyond the bitrate, but I limited the experiment to this model only.

I will remake tests with a Q4_K output on other models to see if the gain is worth it.